src/entities/dto/create-data-target.dto.ts

```diff
@@ -58,6 +58,15 @@ export class CreateDataTargetDto {
     @ApiProperty({ required: false, default: "", example: null })
     authorizationHeader: string;
 
+    @ApiProperty({ required: false })
+    tokenEndpoint?: string;
+
+    @ApiProperty({ required: false })
+    clientId?: string;
+
+    @ApiProperty({ required: false })
+    clientSecret?: string;
```

Comment:
Be aware that client ID and Secret are stored in plain text in the database, which may not be best practice. Make sure this is acceptable for your case.
Is it possible for you to abstract storing and retrieving the token into an interface and implementation? For instance, others may want to make an implementation which stores the secret in, and retrieves it from, Azure Key Vault, Kubernetes secrets etc.

Reply:
For the sake of this PR, we are fine with storing client secrets in plain text in the database, but I have refactored the code so it will be possible to provide an alternative implementation with something more sophisticated